crates/sui-json-rpc-types/src/sui_object.rs

```diff
@@ -31,11 +31,27 @@ use sui_types::object::{Data, MoveObject, Object, ObjectFormatOptions, ObjectRea
 use crate::{Page, SuiMoveStruct, SuiMoveValue};
 
 #[derive(Serialize, Deserialize, Debug, JsonSchema, Clone, PartialEq, Eq)]
-#[serde(tag = "status", content = "details", rename = "ObjectRead")]
-pub enum SuiObjectResponse {
-    Exists(SuiObjectData),
-    NotExists(ObjectID),
-    Deleted(SuiObjectRef),
+pub struct SuiObjectResponse {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub data: Option<SuiObjectData>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error: Option<SuiObjectResponseError>,
+}
```

Comment:
Per @gegaowp 's suggenstion, we can also do

```
pub enum SuiObjectResponse {
  data(SuiObjectData),
  error(suiObjectResponseError)
}
```
it does not change the client interface because of the `skip_serializing_if option is none`, but may be easier to handle in rust.

Reply:
yeah the main point is that, as the response should always either data or error, so that on Rust end, the consumer of this response do not need to handle cases when both are Some or both are None.

Reply:
that should not happen, each object has at most 1 KV pair in storage on FN, it is either deleted or still exists.
same thing on indexer, each object has at most 1 row, either with deleted status or existing status.